src/main/java/com/powsybl/balances_adjustment/util/CountryArea.java

```diff
@@ -6,61 +6,115 @@
+        resetCache();
     }
 
-    public Country getCountry() {
-        return country;
+    public List<Country> getCountries() {
+        return countries;
     }
 
     @Override
-    public List<VoltageLevel> getAreaVoltageLevels(Network network) {
-        return network.getVoltageLevelStream()
-                .filter(voltageLevel -> voltageLevel.getSubstation().getCountry().isPresent())
-                .filter(voltageLevel -> voltageLevel.getSubstation().getCountry().get().equals(country))
-                .collect(Collectors.toList());
+    public double getNetPosition(Network network) {
+        cacheAreaBorders(network);
+        double areaNetPostion = 0.;
+        for (DanglingLine danglingLine : danglingLineBordersCache) {
+            areaNetPostion += getLeavingFlow(danglingLine);
+        }
+        for (Line line : lineBordersCache) {
+            areaNetPostion += getLeavingFlow(line);
+        }
+        for (HvdcLine hvdcLine : hvdcLineBordersCache) {
+            areaNetPostion += getLeavingFlow(hvdcLine);
+        }
+        return areaNetPostion;
     }
 
     @Override
-    public List<BorderDevice> getBorderThreeWindingTransformer(Network network, List<VoltageLevel> areaVoltageLevels) {
-        return new ArrayList<>();
+    public void resetCache() {
+        danglingLineBordersCache = new ArrayList<>();
+        lineBordersCache = new ArrayList<>();
+        hvdcLineBordersCache = new ArrayList<>();
     }
 
-    @Override
-    public String getName() {
-        return country.getName();
+    private void cacheAreaBorders(Network network) {
+        if (danglingLineBordersCache.isEmpty()) {
+            danglingLineBordersCache = network.getDanglingLineStream()
+                    .filter(this::isAreaBorder)
+                    .collect(Collectors.toList());
+        }
+        if (lineBordersCache.isEmpty()) {
+            lineBordersCache = network.getLineStream()
+                    .filter(this::isAreaBorder)
+                    .collect(Collectors.toList());
+        }
+        if (hvdcLineBordersCache.isEmpty()) {
+            hvdcLineBordersCache = network.getHvdcLineStream()
+                    .filter(this::isAreaBorder)
+                    .collect(Collectors.toList());
+        }
     }
 
-    @Override
-    public boolean equals(Object object) {
-        if (object == this) {
-            return true;
+    private boolean isAreaBorder(DanglingLine danglingLine) {
+        Country country = danglingLine.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null);
+        return countries.contains(country);
+    }
+
+    private boolean isAreaBorder(Line line) {
+        Country countrySide1 = line.getTerminal1().getVoltageLevel().getSubstation().getCountry().orElse(null);
+        Country countrySide2 = line.getTerminal2().getVoltageLevel().getSubstation().getCountry().orElse(null);
+        if (countrySide1 == null || countrySide2 == null) {
+            return false;
         }
```

Comment:
Is this test really necessary: in the previous method, null seems to be handled properly.